monetio/models/raqms.py

```diff
@@ -52,21 +60,36 @@ def open_mfdataset(fname, *, surf_only=False):
             "in netCDF format."
             "Do not mix and match file types."
         )
-
     ds = xr.open_mfdataset(names, concat_dim="time", drop_variables=["theta"], combine="nested")
-    ds = _fix(ds, surf_only=surf_only)
+    if var_list is not None:
+        var_list.extend(["lat", "lon", "IDATE", "Times", "psfc", "delp", "pdash", "ttheta"])
+        ds = ds[var_list]
+    ds = _fix(ds, surf_only=surf_only, convert_to_ppb=convert_to_ppb)
 
     return ds
 
 
-def _fix(ds, *, surf_only):
+def _fix(ds, *, surf_only, convert_to_ppb):
     ds = _fix_grid(ds)
     ds = _fix_time(ds)
     ds = _fix_pres(ds)
 
     if surf_only:
         ds = ds.isel(z=0).expand_dims("z")
 
+    if convert_to_ppb:
+        for i in ds.variables:
+            if "units" in ds[i].attrs:
+                if ds[i].attrs["units"] == "ppv":
+                    ds[i] *= 1e9
```

Comment:
This seems to be causing an issue. In CI got `FAILED tests/test_raqms.py::test_surf_only[open_dataset] - ValueError: output array is read-only` for this line.

We could try it this way instead:
```suggestion
                    with xr.set_options(keep_attrs=True):
                        ds[i] = ds[i] * 1e9
```

Reply:
Good to know that about CI. Will keep that in mind for future commits.

Reply:
It worked the original way on my machine. And in that run in CI it worked in Python 3.9 but failed in 3.10 and 3.11. I looked a bit at the env lists in the runs but nothing jumped out at me so far...